public/pages/workflow_detail/agentic_search/configure_flow/tool_forms/query_planning_tool.tsx

```diff
@@ -348,3 +385,41 @@ export function QueryPlanningTool(props: QueryPlanningToolProps) {
     </>
   );
 }
+
+// attempt to parse the upstream connector details and try to derive the relevant response filter.
+// Follows very similar logic to getRelevantInterface() in agent_advanced_settings.tsx.
+// Currently, only OpenAI and Bedrock Claude have native response filtering support.
+export function getRelevantResponseFilter(
+  model: Model,
+  connectors: ConnectorDict
+): RESPONSE_FILTER_TYPE | undefined {
+  // A connector can be defined within a model, or a reference to a standalone connector ID.
+  const connector = !isEmpty(getIn(model, 'connector', {}))
+    ? getIn(model, 'connector', {})
+    : getIn(connectors, getIn(model, 'connectorId', ''), {});
+
+  const connectorModel = getIn(connector, 'parameters.model', '') as string;
+  const connectorServiceName = getIn(
+    connector,
+    'parameters.service_name',
+    ''
+  ) as string;
+  const remoteInferenceUrl = getIn(connector, 'actions.0.url', '') as string;
+
+  if (connectorModel.includes('gpt') || remoteInferenceUrl.includes('openai')) {
+    return RESPONSE_FILTER_TYPE.OPENAI;
+  } else if (
+    connectorModel.includes('claude') &&
+    connectorServiceName.includes('bedrock')
+  ) {
+    return RESPONSE_FILTER_TYPE.BEDROCK_CLAUDE;
+  } else if (
+    connectorModel.includes('deepseek') &&
```

Comment:
this check can be part of else block

Reply:
sure- it was there to be placeholders for future additions, but not needed. will remove